### PR TITLE
Instruction count

### PR DESCRIPTION
**Pull request by paulmure**

### CODE REVIEW

File: src/instruction_counter.rs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    if paths.len() != 1 {
+        panic!("found multiple dep files");
+    }
+    paths[0].to_str().unwrap().to_owned()
+}
+
+/// Read the output of llc from disc
+fn read_llc_output(mut bc_path: String) -> String {
+    let len = bc_path.len();
+    bc_path.truncate(len - 2);
```
Comment (paulmure) [inline src/instruction_counter.rs:25]:
```suggestion
   // change the file extension from ".bc" to ".s"
    bc_path.truncate(len - 2);
```

File: src/instruction_counter.rs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+/// Count the number of machine instructions corresponding to the current path
+pub fn count_instructions<'p, B: Backend>(
+    disassembly: &Disassem,
+    state: &State<'p, B>,
+) -> Result<(String, usize), String> {
+    // matches any line that is a machine instruction
+    let instr_re = Regex::new(r"^(\s*)([^@_\s\.])(.*)$").unwrap();
+    // matches the start of a function or bb
+    let bb_or_func_re = Regex::new(r"(^_.+:$)|(^@\s*%bb\.\d+:.*$)").unwrap();
```
Comment (hudson-ayers) [inline src/instruction_counter.rs:67]: If we are counting instructions along a lot of paths we may want to compile these regex's once using lazy_static (check out the regex docs), but this is fine for now. Compiling is much slower than matching so it saves a reasonable amount of time to only compile each regex once.

File: src/instruction_counter.rs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+    let mut res = String::new();
+    let mut num_instrs = 0;
+
+    // TODO: can control flow leave a bb in the middle?
```
Comment (hudson-ayers) [inline src/instruction_counter.rs:72]: Yeah, lets borrow the logic from `pretty_path_llvm_instrs()` that checks for this and just panic if we see it happen. Then if it does happen in practice we can come back and figure out how to address it.

**Review by hudson-ayers — APPROVED** [review]
This looks great overall, I am gonna go ahead and merge it as it looks useful already. You can just make the edits suggested in my two comments as part of a later PR.

Thanks!

**Pull request merged**